packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -358,9 +358,10 @@ contract Bridge is EssentialContract, IBridge {
         whenNotPaused
         sameChain(message.destChainId)
     {
-        // If the gasLimit is set to 0 or isLastAttempt is true, the caller must
-        // be the message.owner.
-        if (message.gasLimit == 0 || isLastAttempt) {
+        // If isLastAttempt is true, the caller must be the message.owner.
+        // TODO(Brecht): why not allow anyone to call when message.gasLimit == 0?
+        // There is no fee to be earned, and there won't be any gas limit anyway.
+        if (isLastAttempt) {
```

Comment:
Explained in the comment, any reason to check for the `message.gasLimit == 0` case before?

Reply:
My understanding is kinda changed, as Brecht highlighted.
If `message.gasLimit == 0 `-> we allow only the owner to call this. But why ? No fees to be earned by anyone so it is an extra check but might be unnecessary. What is the reason noone can call this retry on behalf of the user, as he/she does not gain any fees ?
And user can still decide if this is the `lastAttempt`, he/she can just call it with true and message status will be permanently set.

Reply:
> we allow only the owner to call this. But why 
Because in some scenarios the users want to make the call when the timing is right, it has nothing to do with fees, it's about the timing.

Reply:
> Because in some scenarios the user wants to make the call when the timing is right, it has nothing to do with fees, it's about the timing.

What kind of timing, or why timing ? You mean timing 'when' cheap - no congestion ? If so, still not hurt if someone else claims for us when txn would be expensive for us and we wait till the proper time. In the meantime if anyone claims for us (free) it is OK - in my opinion.

Or you mean some other kind of timing reason ?

Reply:
Timing is indeed an interesting case I hadn't thought of. Though this would only be for retrying messages that already failed a first time, so seems like if it was timing sensitive, a retry as fast as possible afterwards would be okay.

But fair enough, can't hurt to keep it for this potential edge case!